synapse/appservice/scheduler.py

```diff
@@ -212,7 +227,40 @@ async def _send_request(self, service: ApplicationService) -> None:
                 ]
                 del all_to_device_messages[:MAX_TO_DEVICE_MESSAGES_PER_TRANSACTION]
 
-                if not events and not ephemeral and not to_device_messages_to_send:
+                # Consolidate any pending device list summaries into a single, up-to-date
+                # summary.
+                # Note: this code assumes that in a single DeviceLists, a user will
+                # never be in both "changed" and "left" sets.
```

Comment:
I think the code below will actually work fine if the sets are not disjoint, with `left` winning...which might not be what you want. 👍

Reply:
The spec [doesn't particularly say](https://spec.matrix.org/v1.2/client-server-api/#extensions-to-sync-1) what a client should do if a user ends up in both lists, though I would assume that `left` should win (as if a user updated their list, but they've now left all e2ee rooms with you, then why do you care?).

...So maybe it's OK to pass a device list update with a user with both lists anyways?